src/VisualStudio/Core/Def/StackTraceExplorer/StackFrameViewModel.cs

```diff
@@ -0,0 +1,310 @@
+            _workspace = workspace;
+            _streamingPresenter = streamingPresenter;
+        }
+
+        public override bool ShowMouseOver => true;
+
+        public void NavigateToClass()
+        {
+            var cancellationToken = _threadingContext.DisposalToken;
+            Task.Run(() => NavigateToClassAsync(cancellationToken), cancellationToken);
+        }
+
+        public async Task NavigateToClassAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                var symbol = await GetSymbolAsync(cancellationToken).ConfigureAwait(false);
+
+                if (symbol is not { ContainingSymbol: not null })
+                {
+                    // Show some dialog?
+                    return;
+                }
+
+                // Use the parent class instead of the method to navigate to
+                symbol = symbol.ContainingSymbol;
+
+                var sourceLocation = symbol.Locations.FirstOrDefault(l => l.IsInSource);
+                if (sourceLocation is null || sourceLocation.SourceTree is null)
+                {
+                    // Show some dialog?
+                    return;
+                }
+
+                var navigationService = _workspace.Services.GetService<IDocumentNavigationService>();
+                if (navigationService is null)
+                {
+                    return;
+                }
+
+                // While navigating do not activate the tab, which will change focus from the tool window
+                var options = _workspace.Options
+                        .WithChangedOption(new OptionKey(NavigationOptions.PreferProvisionalTab), true)
+                        .WithChangedOption(new OptionKey(NavigationOptions.ActivateTab), false);
+
+                var document = _workspace.CurrentSolution.GetRequiredDocument(sourceLocation.SourceTree);
+
+                await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+                navigationService.TryNavigateToSpan(_workspace, document.Id, sourceLocation.SourceSpan, options, cancellationToken);
+            }
+            catch (Exception ex) when (FatalError.ReportAndCatchUnlessCanceled(ex, cancellationToken))
+            {
+            }
+        }
+
+        public void NavigateToSymbol()
+        {
+            var cancellationToken = _threadingContext.DisposalToken;
+            Task.Run(() => NavigateToMethodAsync(cancellationToken), cancellationToken);
+        }
+
+        public async Task NavigateToMethodAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                var symbol = await _frame.ResolveSymbolAsync(_workspace.CurrentSolution, cancellationToken).ConfigureAwait(false);
+
+                if (symbol is null)
+                {
+                    // Show some dialog?
+                    return;
+                }
+
+                var success = GoToDefinitionHelpers.TryGoToDefinition(
```

Comment:
Curious why this uses a different navigation mechanism?

Reply:
I forgot to update the other one...